tree/treeplayer/src/TTreeReaderArray.cxx

```diff
@@ -122,7 +122,11 @@ namespace {
       virtual size_t GetSize(ROOT::Detail::TBranchProxy* proxy) {
          TVirtualCollectionProxy *myCollectionProxy = GetCP(proxy);
          if (!myCollectionProxy) return 0;
-         TVirtualCollectionProxy::TPushPop ppRaii(myCollectionProxy, proxy->GetWhere());
+         // Here we do not use a RAII but we empty the proxy to then fill it.
+         // This is done because we are returning a pointer and we need to keep
+         // alive the memory it points to.
```

Comment:
I don't understand this pattern.   If there is nested use of the collection proxy this would change the state that the caller is expected ...

What is the memory that need to be kept alive?  How is adding it to the collection proxy helps?

Reply:
mistake. This shall not be there.